### PR TITLE
store/tikv: re-add peer

### PR DESCRIPTION
**Pull request by disksing**
@qiuyesuifeng @siddontang @tiancaiamao @shenli PTAL

### CODE REVIEW

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -57,69 +57,72 @@ func (c *RegionCache) DropRegion(regionID uint64) {
...
 
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	delete(c.regions, regionID)
 
-	if curStoreIdx == -1 || err != nil {
-		// Can't find the store in cache, or error occurs when loading
+	if peer == nil || err != nil {
+		// Can't find the peer in cache, or error occurs when loading
 		// store from PD.
```
Comment (shenli) [inline store/tikv/region_cache.go]: store->peer

File: store/tikv/region_cache.go [hunk 2]
```diff
@@ -171,9 +175,10 @@ func (c *RegionCache) loadRegion(key []byte) (*Region, error) {
 
 // Region store region info. Region is a readonly class.
```
Comment (shenli) [inline store/tikv/region_cache.go]: stores

File: store/tikv/region_cache.go (hunk 1 above)
Comment (disksing) [inline store/tikv/region_cache.go]: @shenli It's not a typo here. `err == nil` means L107 failed load Store from PD.

**Comment by shenli** [comment]
LGTM

File: store/tikv/kv.go [hunk 3]
```diff
@@ -185,15 +185,15 @@ func (s *tikvStore) SendKVReq(req *pb.Request, region *requestRegion) (*pb.Respo
...
+			log.Warnf("send tikv request error: %v, try next peer later", err)
+			s.regionCache.NextPeer(region.GetID())
 			continue
 		}
 		if regionErr := resp.GetRegionError(); regionErr != nil {
 			// Retry if error is `NotLeader`.
 			if notLeader := regionErr.GetNotLeader(); notLeader != nil {
 				log.Warnf("tikv reports `NotLeader`: %s, retry later", notLeader.String())
-				s.regionCache.UpdateLeader(notLeader.GetRegionId(), notLeader.GetLeaderStoreId())
+				s.regionCache.UpdateLeader(notLeader.GetRegionId(), notLeader.GetLeader().GetId())
```
Comment (siddontang) [inline store/tikv/kv.go]: GetLeader() can return nil here?

**Comment by siddontang** [comment]
Rest LGTM

File: store/tikv/kv.go (hunk 3 above)
Comment (disksing) [inline store/tikv/kv.go]: @siddontang No matter. Protobuf checks if m==nil, so it will return 0 if GetLeader returns nil.

**Pull request merged**